### PR TITLE
SDLManagerDelegate support AudioStreamingState and SystemContext

### LINKED ISSUE #569: [SDL 0032] Delegate Support for all onHMIStatus State Changes
Proposal: [Delegate Support for all onHMIStatus State Changes](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0032-on-hmi-status-state-changes.md)

> This proposal is to add in support to inform developers of additional state changes with onHMIStatus. Currently, [SDLManagerDelegate](https://github.com/smartdevicelink/sdl_ios/blob/master/SmartDeviceLink/SDLManagerDelegate.h) only supports SDLHMIStatus changes, and is missing callbacks for SDLAudioStreamingState as well as SDLSystemContext.

Review: smartdevicelink/sdl_evolution#101

SDLC Steering Committee Decision:

> The Steering Committee fully agreed to accept this proposal.

### PR DESCRIPTION
**Pull request by asm09fsu**
Fixes #569 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests were written for these changes.

### Summary
This PR implements [SDL-0032](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0032-on-hmi-status-state-changes.md).

> This proposal is to add in support to inform developers of additional state changes with `onHMIStatus`. Currently, [SDLManagerDelegate](https://github.com/smartdevicelink/sdl_ios/blob/master/SmartDeviceLink/SDLManagerDelegate.h) only supports `SDLHMIStatus` changes, and is missing callbacks for `SDLAudioStreamingState` as well as `SDLSystemContext`.

##### Enhancements
* Added in support for system context and audio streaming state change delegate callbacks.

### CODE REVIEW

File: SmartDeviceLink/SDLLifecycleManager.h [hunk 1]
```diff
@@ -9,8 +9,10 @@
 #import "SDLNotificationConstants.h"
 #import <Foundation/Foundation.h>
 
+#import "SDLAudioStreamingState.h"
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLifecycleManager.h]: These can be `@class`, right?

File: SmartDeviceLink/SDLLifecycleManager.m [hunk 2]
```diff
@@ -186,7 +186,13 @@ - (void)sdl_stopManager:(BOOL)shouldRestart {
     self.registerResponse = nil;
     self.lastCorrelationId = 0;
     self.hmiLevel = nil;
-
+    
+    // New Start
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLLifecycleManager.m]: ???

File: SmartDeviceLink/SDLLifecycleManager.h (hunk 1 above)
Comment (asm09fsu) [inline SmartDeviceLink/SDLLifecycleManager.h]: These are not classes anymore, remember? They are enums.

File: SmartDeviceLink/SDLLifecycleManager.m (hunk 2 above)
Comment (asm09fsu) [inline SmartDeviceLink/SDLLifecycleManager.m]: Resolved.

**Pull request merged**